test/articles.spec.js

```diff
@@ -39,30 +40,32 @@ describe('/POST article', () => {
             done();
           });
     });
-    // it('it should POST a article ', (done) => {
-    //     let article = {
-    //         title: "The Lord of the Rings",
-    //         author: "J.R.R. Tolkien",
-    //         content: "The lord of the rings"
-    //     }
-    //       chai.request(server)
-    //       .post('/api/v1/articles')
-    //       .send(article)
-    //       .end((err, res) => {
-    //             res.should.have.status(200);
-    //             res.body.should.be.a('object');
-    //             res.body.should.have.property('data');
-    //             res.body.data.should.have.property('article');
-    //             res.body.data.article.should.have.property('title');
-    //             res.body.data.article.should.have.property('author');
-    //             res.body.data.article.should.have.property('content');
-    //         done();
-    //       });
-    // });
+    it('it should POST an article ', (done) => {
+        let token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2MWU2OWFlOTFmMDEyZjg2MDE4ODQwNTMiLCJuYW1lIjoiRXJpYyIsInBhc3N3b3JkIjoiJDJiJDEwJGJtSUtSUzN6UU56Y1pMaG0wM3NUZk9sVmFWcG16N2J5cnRydnNNMEV6WUguTlQuVXJqM25HIiwiX192IjowLCJpYXQiOjE2NDI1MTYzMjcsImV4cCI6MTY0MjUxOTkyN30.9QKmLYzEjrIxEndJ7HvAKMMwYZwFjgDeMupw2nTKBes"
+        let article = {
+            title: "The Lord of the Rings",
+            author: "J.R.R. Tolkien",
+            content: "The lord of the rings"
+        }
+          chai.request(server)
+          .post('/api/v1/articles')
+          .auth(token, { type: 'bearer' })
+          .send(article)
+          .end((err, res) => {
+                res.should.have.status(201);
+                res.body.should.be.a('object');
+                res.body.should.have.property('data');
+                res.body.data.should.have.property('article');
+                res.body.data.article.should.have.property('title');
+                res.body.data.article.should.have.property('author');
+                res.body.data.article.should.have.property('content');
+            done();
+          });
+    });
 });
 describe('/GET/:id article', () => {
     it('it should GET the article by the given id', (done) => {
-        let id = "61e4014690b12add9ba7b332"
+        let id = "61e55e5379904e9bd63020af"
```

Comment:
Same as for the token. We can get these from request responses. Check this snippet. Not sure if its good testing practice, but works so great.
```
  describe("User login before creating a post", async () => {
    it("It should return a token for the logged in user", async () => {
      let userInfo = {
        password: "Your password of choice",
        email: "A valid email or username",
      };
      const userResult = chai.request(app)

        .post(apiRoute + "/accounts/login")
        .send(userInfo);
      expect(userResult).to.have.status(200);
      token = userResult.body.data.token;
    });
  });
  describe("Creating a new post after login", async () => {
    it("Should return a newly created post", async () => {
      const postCreation = await chai
        .request(app)
        .post(apiRoute + "/blogs")
        .send(post)
        .set("Authorization", "Bearer " + token);
      expect(postCreation).to.have.status(200);
      expect(postCreation.body.data).to.have.property("slug");
      postId = postCreation.body.data._id;
    });
  });
```

Reply:
I changed this too. I am getting the id from the response. Will push my code and create another PR